src/ProjectOrigin.VerifiableEventStore/Services/Repository/InMemory/InMemoryRepository.cs

```diff
@@ -69,29 +73,60 @@ public Task FinalizeBlock(BlockHash hash, ImmutableLog.V1.BlockPublication publi
         }
     }
 
-    public Task<ImmutableLog.V1.Block?> GetBlock(TransactionHash transactionHash)
+    public Task<Registry.V1.Block?> GetBlock(TransactionHash transactionHash)
     {
         lock (_lockObject)
         {
             var e = _events.SingleOrDefault(x => x.TransactionHash == transactionHash);
 
             if (e is null)
-                return Task.FromResult<ImmutableLog.V1.Block?>(null);
+                return Task.FromResult<Registry.V1.Block?>(null);
 
             var index = _events.IndexOf(e);
 
             var block = _blocks.Values.SingleOrDefault(x => x.FromTransaction <= index && index <= x.ToTransaction);
 
             if (block is null)
-                return Task.FromResult<ImmutableLog.V1.Block?>(null);
+                return Task.FromResult<Registry.V1.Block?>(null);
 
-            var result = new ImmutableLog.V1.Block
+            var result = new Registry.V1.Block
             {
                 Header = block.Header,
                 Publication = block.Publication
             };
 
-            return Task.FromResult<ImmutableLog.V1.Block?>(result);
+            return Task.FromResult<Registry.V1.Block?>(result);
+        }
+    }
+
+    public Task<IList<Block>> GetBlocks(int skip, int take, bool includeTransactions)
+    {
+        lock (_lockObject)
+        {
+            var data = _blockList.Skip(skip).Take(take).Select(x =>
+            {
+                var block = new Block
+                {
+                    Header = x.Header,
+                    Publication = x.Publication,
+                    Height = _blockList.IndexOf(x) + 1
+                };
```

Comment:
Vi har en O(n^2) her. Hvis _blocklisten vokser, så er der potentiel flaskehals her.

Reply:
The inMemory is only for tests purposes so I dont believe it as very relevant.